anisoap/representations/radial_basis.py

```diff
@@ -1,4 +1,87 @@
+import warnings
+
 import numpy as np
+from equistore.core import TensorMap
+from scipy.special import gamma
+
+
+def inverse_matrix_sqrt(matrix: np.array):
+    """
+    Returns the inverse matrix square root.
+    The inverse square root of the overlap matrix (or slices of the overlap matrix) yields the
+    orthonormalization matrix
+    Args:
+        matrix: np.array
+            Symmetric square matrix to find the inverse square root of
+
+    Returns:
+        inverse_sqrt_matrix: S^{-1/2}
+
+    """
+    if not np.allclose(matrix, matrix.T):
+        raise ValueError("Matrix is not hermitian")
+    eva, eve = np.linalg.eigh(matrix)
+
+    if (eva < 0).all():
+        raise ValueError(
+            "Matrix is not positive semidefinite. Check that a valid gram matrix is passed."
+        )
```

Comment:
I'm not sure these checks belong here -- do we ever worry about them being triggered?

Reply:
Hmm in its current usage these checks will not be triggered. However, these are general purpose util functions that could theoretically be (mis)used elsewhere. 

So, should it be the function's role to verify that the matrix is symmetric positive semidefinite, or should such a verification be done outside the function?